Library/OcDebugLogLib/OcLog.c

```diff
@@ -140,18 +146,129 @@ GetLogPath (
   return LogPath;
 }
 
+STATIC
 EFI_STATUS
-EFIAPI
-OcLogAddEntry  (
-  IN OC_LOG_PROTOCOL    *OcLog,
-  IN UINTN              ErrorLevel,
-  IN CONST CHAR8        *FormatString,
-  IN VA_LIST            Marker
+GetLogPrefix (
+  IN   CONST CHAR8  *FormatString,
+  OUT  CHAR8        *Prefix
   )
 {
-  EFI_STATUS                  Status;
+  UINTN    MaxLength;
+  UINTN    Index;
+  CHAR8    Curr;
 
-  OC_LOG_PRIVATE_DATA         *Private;
+  ASSERT (FormatString != NULL);
+  ASSERT (Prefix != NULL);
+
+  //
+  // If FormatString just starts with colon, it must be illegal.
+  //
+  if (*FormatString == ':') {
+    return EFI_NOT_FOUND;
+  }
+
+  MaxLength = MIN (AsciiStrLen (FormatString), OC_LOG_PREFIX_CHAR_MAX);
+  for (Index = 1; Index < MaxLength; ++Index) {
```

Comment:
This is going to result in prefixes which are OC_LOG_PREFIX_CHAR_MAX long failing, right? I guess you want them to succeed.

Reply:
No. Either there is a very short string which has the length of `AsciiStrLen (FormatString)`, or we just match up to `OC_LOG_PREFIX_CHAR_MAX (8)` characters.

Reply:
Yes, but if the prefix is 8 chars, and the prefix you are trying to match is 8 chars, then you will match 8 chars here, and then you will hit `if (Index == MaxLength) { return EFI_NOT_FOUND; }` below, I think.

Reply:
https://github.com/acidanthera/OpenCorePkg/commit/ecb0a3096efbb799ca4cfb06207b6dae87693701